### PR TITLE
Don't expose BackgroundThread

### PR DESCRIPTION
**Pull request by lexs**
This fixes the error introduced in #1741.

### CODE REVIEW

**Comment by lexs** [comment]
I didn't update the tests in RAP because I didn't have time to understand them.

However I think we really need a simple Robolectric sample to catch problems like these, this was a compile error.

File: robolectric-shadows/shadows-core/src/main/resources/org/robolectric/shadows/ShadowBackgroundThread.java.vm [hunk 1]
```diff
@@ -9,7 +9,7 @@ import org.robolectric.annotation.Resetter;
 import org.robolectric.internal.Shadow;
 import org.robolectric.util.ReflectionHelpers;
 
-@Implements(value = BackgroundThread.class, inheritImplementationMethods = true)
+@Implements(value = BackgroundThread.class, isInAndroidSdk = false, inheritImplementationMethods = true)
```
Comment (jongerrish) [inline robolectric-shadows/shadows-core/src/main/resources/org/robolectric/shadows/ShadowBackgroundThread.java.vm]: I'm sorry that slipped out of the original PR. I did have that in a version I was working on. Thanks for catching this!

File: robolectric-processor/src/main/java/org/robolectric/annotation/processing/RobolectricModel.java [hunk 2]
```diff
@@ -227,6 +227,11 @@ public void prepare() {
       registerType(solidType);
     }
 
+    for (Map.Entry<TypeElement,TypeElement> entry : getResetterShadowTypes().entrySet()) {
```
Comment (jongerrish) [inline robolectric-processor/src/main/java/org/robolectric/annotation/processing/RobolectricModel.java]: I'll merge this now as it fixes the breakage I introduced, would you mind adding tests in a follow up PR though?

**Comment by jongerrish** [comment]
Thanks for the fix!

**Comment by erd** [comment]
> However I think we really need a simple Robolectric sample to catch problems like these, this was a compile error.

You can run your local Robolectric build against the tests in the `robolectric-samples` repo.

**Comment by lexs** [comment]
> You can run your local Robolectric build against the tests in the robolectric-samples repo.
Sure, but I was thinking of something that was part of the regular build, so #1741 would have failed during the test run.

**Comment by erd** [comment]
I've wanted something like that for a long time. I wish Travis would let you trigger builds of dependent projects, so we could build the samples with every new snapshot build. I'm open to suggestions if you can think of a way to do this.

**Pull request merged**